### ISSUE TITLE
known-third-party does not work with namespace packages

### ISSUE THREAD

**Issue by joshuaavalon** [PyCQA/pylint#2691]
### Steps to reproduce
I am using [namespace package](https://packaging.python.org/guides/packaging-namespace-packages/). I have a `example.foo` which is imported in `example.bar`

When I use absolute import, it throws errors like `wrong-import-order` and `ungrouped-imports`. For example,
```python
from example.foo import foo
from ruamel.yaml import YAML

from example.bar.base import BarBase
```

I add `example.foo` to `known-third-party` but it does not work. Adding `example` works but it still create false positive that `example.bar` is not grouped with `example.foo`.

### Current behavior
known-third-party work with namespace packages

### Expected behavior
It has not effect

### pylint --version output
```
pylint 2.2.2
astroid 2.0.2
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:06:47) [MSC v.1914 32 bit (Intel)]
```

**Comment by PCManticore** [comment]
@joshuaavalon Thanks for reporting the issue. I'm not sure it's valid though, `known-third-party` seems to work when passing `example`. From your example, `ungrouped-imports` is expected, that check expects all the imports from the same module to be arranged near each other.

**Comment by webknjaz** [comment]
@PCManticore I just hit the same problem. We have an implicit namespace with various projects in it, and one of them is a “platform”. So naturally, the platform one should go before most things. It's like a framework. I made a separate section in `isort` like `known_platforms = ecosystem_namespace.platform` and I'd expect that the local in-project imports would be towards the end of the list, while the platform would be right after stdlib, with third-party modules in-between.

So something like this should be allowed:
```python
from __futures__ import annotations

import os

import ecosystem_namespace.platform

import cheroot

from ecosystem_namespace.local_project import smth
```

And `isort` lets me do this when I configure it correctly. PyLint seems to be relying on `isort` for the import order checks. So it's a bit confusing why it would treat `ecosystem_namespace.platform` as first-party (`ecosystem_namespace` really) and not the proper section.

**Comment by webknjaz** [comment]
Filed https://github.com/pylint-dev/pylint/issues/9977 with a longer explanation.
